pkg/asset/installconfig/azure/dns.go

```diff
@@ -139,6 +146,19 @@ func (client *ZonesClient) GetAllPublicZones() (map[string]string, error) {
 	return allZones, nil
 }
 
+// DNSZoneExists validates if a DNS zone exists in the provided resource group
+func (client *ZonesClient) ZoneExists(rgName, zoneName string) bool {
+	ctx, cancel := context.WithTimeout(context.TODO(), 30*time.Second)
+	defer cancel()
+
+	_, err := client.azureClient.Get(ctx, rgName, zoneName)
+	if err != nil {
+		return false
+	}
```

Comment:
curious: should be distinguish between not-found and other errors? Maybe, we can check the status code for 404?

Reply:
Would it be useful to log the error returned here or at the caller?